File: frontend/web/src/api/courses/getCourse.ts
import axiosInstance from "../../utils/AxiosInstance.ts"

const getCourse = async (id: number) => {
    const token = localStorage.getItem('token');

    const { data } = await axiosInstance.get(`/api/courses/${id}`, {
        headers: {
            Authorization: `Bearer ${token}`,
        }
    });

    return data.course;
}

export default getCourse;

File: frontend/desktop/python-client/PyQt5App/api/courses/getCourse.py
import requests
from PyQt5App.utils.Address import address
from PyQt5App.utils.sessionManager import SessionManager

class GetCourse:
    def __init__(self, course_id):
        self.address = address()
        self.sessionManager = SessionManager()
        self.token = self.sessionManager.get_token()
        self.course_id = course_id

    def getCourse(self):
        response = requests.get(
            f"{self.address}/api/courses/{self.course_id}",
            headers={
                "Authorization": f"{self.token}",
            }
        )
        data = response.json()
        return data.get("course")